test/suite/test_checkpoint05.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2020 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_checkpoint05.py
+# Verify that we don't accumulate a lot of checkpoints while a backup
+# cursor is open. WiredTiger checkpoints created after the backup cursor
+# should get deleted as usual.
+
+import time
+import wiredtiger, wttest
+
+def timestamp_str(t):
+    return '%x' % t
+
+class test_checkpoint04(wttest.WiredTigerTestCase):
+    conn_config = 'create,cache_size=100MB,log=(archive=false,enabled=true,file_max=100K)'
+
+    def make_key(self, n):
+        return "Key:" + str(n)
+
+    def make_value(self, n):
+        return "Value:" + str(n)
+
+    def do_updates(self, n):
+        nops = 500
+        ckpt_freq = 10
+
+        cursor = self.session.open_cursor(self.uri, None)
+        for i in range(n * nops, (n + 1) * nops - 1):
+            cursor[self.make_key(i)] = self.make_value(i)
+            if i % ckpt_freq == 0:
+                self.session.checkpoint(None)
+        cursor.close()
```

Comment:
I think you don't need these 3 functions. I think the test can be simplified. You can just add one simple loop in the beginning to add a little data. Then, since we don't care about the data, after waiting for the wall clock to change, you can just loop forcing N checkpoints.

Reply:
These 3, `make_key`, `make_value` and `do_updates` can be removed.